spark/src/main/scala/org/apache/spark/sql/comet/CometTakeOrderedAndProjectExec.scala

```diff
@@ -137,6 +138,6 @@ object CometTakeOrderedAndProjectExec extends ShimCometTakeOrderedAndProjectExec
     val exprs = plan.projectList.map(exprToProto(_, plan.child.output))
     val sortOrders = plan.sortOrder.map(exprToProto(_, plan.child.output))
     exprs.forall(_.isDefined) && sortOrders.forall(_.isDefined) && getOffset(plan).getOrElse(
-      0) == 0
+      0) == 0 && supportedSortType(plan, plan.sortOrder)
```

Comment:
`CometTakeOrderedAndProjectExec` will invoke DataFusion `SortExec` operator, so we should also check for it.